File: orderservice/src/main/java/com/orderservice/dto/OrderDTO.java
package com.orderservice.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.time.LocalDate;
import java.util.List;
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class OrderDTO {

    @JsonProperty("user_id")
    @Min(value = 1,message = "user id must be > 1")
    private Long userId;

    private String fullname;

    @JsonProperty("email")
    private String email;

    @JsonProperty("phone_number")
    @NotBlank(message = "phone number cannot be blank")
    @Size(min = 8, message = "phone number must be at least 8 characters")
    private String phoneNumber;

    @JsonProperty("address")
    private String address;

    @JsonProperty("note")
    private String note;

    @JsonProperty("order_date")
    private LocalDate orderDate;

    private String status;

    @JsonProperty("total_money")
    @Min(value = 0,message = "total must be >= 0 ")
    private Float totalMoney;

    @JsonProperty("shipping_method")
    private String shippingMethod;

    @JsonProperty("shipping_address")
    private String shippingAddress;

    @JsonProperty("shipping_date")
    private LocalDate shippingDate;

    @JsonProperty("tracking_number")
    private String trackingNumber;

    @JsonProperty("payment_method")
    private String paymentMethod;

    private Boolean active;

    @JsonProperty("cart_items")
    List<CartItemDTO> cartItems;
}

File: frontEnd/src/app/dto/order/order.dto.ts
export class OrderDTO {
  user_id: number;
  
  fullname: string;
  
  phone_number: string;
    
  address: string;
  
  status: string;

  note: string;
    
  total_money: number;
  
  shipping_method: string;

  shipping_address: string;
  
  payment_method: string;

  shipping_date: Date;

  tracking_number: string;
  
  cart_items: { product_id: number, quantity: number }[]; 
  
  constructor(data: any) {
    this.user_id = data.user_id;
    this.fullname = data.fullname;
    this.phone_number = data.phone_number;
    this.address = data.address;
    this.note = data.note;
    this.total_money = data.total_money;
    this.shipping_method = data.shipping_method;
    this.payment_method = data.payment_method;
    this.cart_items = data.cart_items;
    this.status = data.status;
    this.shipping_date = data.shipping_date;
    this.tracking_number = data.tracking_number;
    this.shipping_address = data.shipping_address;
  }
}
  
  